### ISSUE TITLE
Add "Dasharo Tools Suite (Beta)" to iPXE menu entry

### ISSUE THREAD

**Issue by macpijan** [Dasharo/dasharo-issues#1041]
Add new menu next to: https://github.com/Dasharo/dasharo-blobs/blob/main/dasharo/dasharo.ipxe#L23

That would boot from: https://boot.dasharo.com/dts/dts-rc.ipxe where releases from `develop` branch land.

**Comment by tlaurion** [comment]
An idea would be to have references to latest ipxe kernel so that Heads could download and boot from it. Just a thought.

**Comment by DaniilKl** [comment]
> Adding (beta) entry which boots into much older RC image than the latest release, seems very much pointless.

I agree.

**Comment by Houmgaor** [comment]
I'd like to claim this one with [dasharo-blobs#53](https://github.com/Dasharo/dasharo-blobs/pull/53). Not particularly technical, but I did full testing and validation.

**Comment by iwanicki92** [comment]
> But this seems not updated anymore. Are we going to update it, or we do not plan to?

Fixed by https://github.com/Dasharo/meta-dts/commit/3f937a1917fa68768e0726bea6c7dbc50a4fd88a (https://github.com/Dasharo/dasharo-issues/issues/1754)
`dts-rc.ipxe` will point to newest release, either `rc` or normal release.

> Maybe I will transform this issue then into adding (nightly) entry.

The problem with nightly builds is that that it still uses `main` branch of https://github.com/Dasharo/dts-configs by default, while `develop` branch of meta-dts (nightly) is tested on `develop` branch of `dts-configs`.
It can be worked around manually, by entering shell in DTS, exporting [`DTS_CONFIG_REF`](https://github.com/Dasharo/dts-scripts/blob/4b0cc017efa7c895b9169f79f999c9278163eec6/include/dts-environment.sh#L204) variable:
`export DTS_CONFIG_REF=refs/heads/develop` and then running DTS UI again: `dts-boot`.
But we might want to change this behaviour to be default one for nightly builds, especially if we want to add it to iPXE menu.

**Comment by DaniilKl** [comment]
> But we might want to change this behaviour to be default one for nightly builds, especially if we want to add it to iPXE menu.

Any ideas how to do so? The first thing that came to my mind was replasing the [DTS_CONFIG_REF value](https://github.com/Dasharo/dts-scripts/blob/4b0cc017efa7c895b9169f79f999c9278163eec6/include/dts-environment.sh#L204) via  the [dts-scripts_git.bb](https://github.com/Dasharo/meta-dts/blob/develop/meta-dts-distro/recipes-dts/dts-scripts/dts-scripts_git.bb) (e.g., by using `sed`). But I am not sure whether it is the optimal solution.

**Comment by m-iwanicki** [comment]
Do it in Makefile if some variable is set (then set it in `EXTRA_OEMAKE` in recipe).
Or you could maybe just add this export to `/etc/profile` so it is set by default.

**Comment by DaniilKl** [comment]
> Do it in Makefile if some variable is set (then set it in `EXTRA_OEMAKE` in recipe). Or you could maybe just add this export to `/etc/profile` so it is set by default.

I agree with the propositions. Then we need to trigger it during Yocto build somehow. We can either do it by passing some variable to kas either by `export The_VARIABLE ...; kas-container` or modifying some `kas/*.yml` file using, for example, `sed` in CI. Or we can add a separate YAML file (for example `kas-develop.yml`) that will be used for building from `develop` branch.

**Comment by yuliuyi717-ux** [comment]
I can take this bounty. Plan:
1) add the new Nightly menu entry in dasharo-blobs () next to the existing DTS entry
2) point it to  behind a build-time toggle so it can be safely enabled
3) update OSFV network boot test/keywords to validate the new menu entry path
4) verify on QEMU and post test evidence in the PR

I’ll split this into focused commits so review is straightforward.

**Comment by yuliuyi717-ux** [comment]
Correction to my previous comment (some inline code got stripped):

1) add the new Nightly menu entry in dasharo-blobs (`dasharo/dasharo.ipxe`) next to the existing DTS entry
2) point it to `https://boot.dasharo.com/dts/nightly.ipxe` behind a build-time toggle so it can be safely enabled
3) update OSFV network boot test/keywords to validate the new menu entry path
4) verify on QEMU and post test evidence in the PR

I will proceed with this plan.

**Comment by yuliuyi717-ux** [comment]
Update: I submitted the implementation in two focused PRs so review can happen per repo:

1) Dasharo/dasharo-blobs#54
- adds `Dasharo Tools Suite (Nightly)` to `dasharo.ipxe`
- chains to `https://boot.dasharo.com/dts/nightly.ipxe`

2) Dasharo/open-source-firmware-validation#1252
- updates `network-boot.robot` menu-order checks for the new entry
- adds a dedicated nightly boot validation test (`PXE004.002`)

Both PRs are linked to this issue and ready for review.

**Comment by redwing509** [comment]
/attempt #1041

Implementation plan:
1. Analyze the existing codebase and understand the requirements
2. Implement the requested changes with clean, tested code
3. Add comprehensive tests covering the acceptance criteria
4. Submit PR with documentation updates if needed

Estimated delivery: within 24 hours.

**Comment by yuliuyi717-ux** [comment]
For clarity, this issue is already covered on my side by the two linked PRs submitted on March 2:

- Dasharo/dasharo-blobs#54 — adds the `Dasharo Tools Suite (Nightly)` menu entry and chains to `https://boot.dasharo.com/dts/nightly.ipxe`
- Dasharo/open-source-firmware-validation#1252 — updates the OSFV network boot checks for the new menu layout and adds dedicated coverage for the Nightly DTS path

That split follows the issue scope directly: one PR for the iPXE menu entry, and one PR for the OSFV test updates mentioned in the bounty description.

If it helps review, I can also add a compact validation summary to either PR, but from an implementation standpoint the requested work for #1041 is already submitted and ready for review.

**Comment by Spina7** [comment]
Opened PRs for this bounty:\n\n- dasharo-blobs: https://github.com/Dasharo/dasharo-blobs/pull/57\n- OSFV tests/keywords: https://github.com/Dasharo/open-source-firmware-validation/pull/1271\n\nThe nightly endpoint currently returns 200 OK, and both PR bodies include local verification.

**Comment by DPS0340** [comment]
Hi @macpijan and @DaniilKl — I would like to take this $100 bounty.

Proposed scope after assignment:
1. Confirm that `dts/nightly.ipxe` is the intended, functional nightly target.
2. Add a clearly labelled Dasharo Tools Suite (Nightly) entry to `dasharo.ipxe`, without changing existing release/RC paths.
3. Extend the OSFV QEMU network-boot coverage and keywords to verify the new menu item and its boot path.
4. Run the relevant QEMU/OSFV validation and include the commands and results in the PR.

I will keep the patch limited to this issue and post progress before opening a PR. Please assign me if this direction is acceptable.

**Comment by DPS0340** [comment]
Update: I found the existing open implementation in Dasharo/dasharo-blobs#53, including the Nightly menu entry. I will not duplicate that work, so please disregard my assignment request and prioritize the existing PR. If the OSFV coverage becomes a separately scoped, unclaimed follow-up after that PR is resolved, I would be happy to discuss it then.

**Comment by Houmgaor** [comment]
> Update: I found the existing open implementation in [Dasharo/dasharo-blobs#53](https://github.com/Dasharo/dasharo-blobs/pull/53), including the Nightly menu entry. I will not duplicate that work, so please disregard my assignment request and prioritize the existing PR. If the OSFV coverage becomes a separately scoped, unclaimed follow-up after that PR is resolved, I would be happy to discuss it then.

Any news with my PR? This thread is a honey-pot for bounty-hunters and bots, and which is quite spamy.

**Comment by jacobo19991** [comment]
Hi! I would like to work on this bounty. I can implement the new Dasharo Tools Suite (Nightly) entry in the iPXE menu, update the relevant OSFV network-boot tests and keywords, and validate the changes with QEMU.

Before starting, I will coordinate with @DaniilKl to confirm that https://boot.dasharo.com/dts/nightly.ipxe points to a functional nightly build.

Could you please assign issue #1041 to me (@jacobo19991)? Once assigned, I will begin and provide progress updates here.

**Comment by DaniilKl** [comment]
[`nightly.ipxe`](https://boot.dasharo.com/dts/nightly.ipxe) has been pointing to a functional build for some time already. There is no reason to ping me for that anymore. IIUC the only thing that is left to be done here is the option in Dasharo Firmware.

**Comment by macpijan** [comment]
> This thread is a honey-pot for bounty-hunters and bots, and which is quite spamy.


@Houmgaor  It is. And not only this one That's why it's troublesome to filter any real/useful contributions.

First of all, if we would follow the process: https://3mdeb.com/bug-bounty/ you should get assigned first to the issue. I am assigning you now.
Your PR looks sane to me and present some manual test resutls - thanks for that.

If you are really willing to complete the issue, we'd need the OSFV test part as well to be submitted as per the issue description and this comment: https://github.com/Dasharo/dasharo-blobs/pull/53#issuecomment-5141030382
